server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -600,30 +604,33 @@ private void returnManagerConnection(ManagerClientService.Client client) {
 
   private HostAndPort startTabletClientService() throws UnknownHostException {
     // start listening for client connection last
-    clientHandler = getThriftClientHandler();
-    Iface rpcProxy = TraceUtil.wrapService(clientHandler);
-    final Processor<Iface> processor;
-    if (getContext().getThriftServerType() == ThriftServerType.SASL) {
-      Iface tcredProxy = TCredentialsUpdatingWrapper.service(rpcProxy, ThriftClientHandler.class,
-          getConfiguration());
-      processor = new Processor<>(tcredProxy);
-    } else {
-      processor = new Processor<>(rpcProxy);
+    TransactionWatcher watcher = new TransactionWatcher(context);
+    clientHandler = newClientHandler(watcher);
+    thriftClientHandler = newTabletClientHandler(watcher);
+
+    try {
+      TProcessor processor = ThriftProcessorTypes.getTabletServerTProcessor(clientHandler,
+          thriftClientHandler, getContext(), getConfiguration());
+      HostAndPort address = startServer(getConfiguration(), clientAddress.getHost(), processor);
+      log.info("address = {}", address);
+      return address;
+    } catch (Exception e) {
+      throw new RuntimeException("Error creating thrift server processor", e);
```

Comment:
Can this be a more specific exception?

Reply:
I was able to remove the Exception declaration from `ThriftProcessorTypes.ProcessorType.getProcessor()` and instead throw an `IllegalArgumentException`. This allowed me to remove this completely in 2bac8d5f30828ada41d60168272b0c6b46802dc5